app/Http/Kernel.php

```diff
@@ -56,7 +58,19 @@ class Kernel extends HttpKernel
         'check-user-restricted' => Middleware\CheckUserRestricted::class,
         'guest' => Middleware\RedirectIfAuthenticated::class,
         'require-scopes' => Middleware\RequireScopes::class,
+        'request-cost' => Middleware\RequestCost::class,
         'throttle' => Middleware\ThrottleRequests::class,
         'verify-user' => Middleware\VerifyUser::class,
     ];
+
+    public function __construct(Application $app, Router $router)
+    {
+        parent::__construct($app, $router);
+
+        // using prepend to add the middlewares before auth
```

Comment:
this should expand more on why they all need to be before auth. And if I understand correctly, failing auth guard will cause a throw and thus the rate limit count increment won't be run?